frontend/src/ContentArea/ContentArea.js

```diff
@@ -1,43 +1,39 @@
 import React from 'react';
-import JSONToHTML from '../JSONHandle';
+import PropTypes from 'prop-types';
+import SingleArticle from './SingleArticle';
+import Topic from './Topic';
+import Root from './Root';
 
 export default class ContentArea extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      innerHTML: undefined,
-    };
+  render() {
+    const { path } = this.props;
+    let view = 'none';
+    const id = JSON.parse(localStorage.getItem('pathTable'))[path] || '';
```

Comment:
I'm not sure it's a good idea to put this in `render`.

Reply:
agree.
We call getItem nine times - perhaps we just call them once at the top level (in App.js) and pass allArticles, pathTable, tree and articleIndex down using props?